src/main/java/com/disi/TravelPoints/service/LandmarkService.java

```diff
@@ -51,6 +54,26 @@ public void add(AddLandmarkRequest request){
         landmarkRepository.save(landmark);
     }
 
+    public Long update(LandmarkDetails request) throws CustomException {
+        Landmark landmark = landmarkRepository.findById(request.getId())
+                .orElseThrow(() -> CustomException
+                        .builder()
+                        .status(HttpStatus.NOT_FOUND)
+                        .message("Landmark not found")
+                        .build());
+
+        landmark.setCategory(request.getCategory());
```

Comment:
Maybe it's better to check if the request.field is not empty and if it's empty to keep the previous value from main entity.